python/prairielearn.py

```diff
@@ -93,7 +94,15 @@ def to_json(v):
             M.append(row)
         return {'_type': 'sympy_matrix', '_value': M, '_variables': s, '_shape': [num_rows, num_cols]}
     elif isinstance(v, pandas.DataFrame):
```

Comment:
I think this should be a boolean flag, but it's easy enough to set this to be the default and modify the test cases. correspondingly.

Reply:
@coatless actually, is the new encoding incompatible with the old? Does it erase data in some cases that might be useful? Not opposed to adding it or anything just curious.

Reply:
I think the issue might be that some courses are expecting the data frame to be passed in the same way if the have custom elements. 

The v1 format stores data in a very different format as columns are separated out from values.

Reply:
Made the legacy encoding the default as you suggested

Reply:
@eliotwrobson I probably would stick with a versioning number, e.g. 1, 2, 3 instead of True/False, since there may be a need to modify further if the upstream pandas bug is resolved.

Reply:
Makes sense. Have changed to the version number and defaulted to 1.